### PR TITLE
fix: need to connect to your stripe account error

### LINKED ISSUE #5262: Wizard Step 1: Should not be possible to save when Stripe is ticked without Stripe connected
The Stripe payment in wizard step 1 can be activated and the event can then be saved. This should not be possible as the user has not connected to the Stripe gateway. It should only be possible to save the event with Stripe ticked on if the user has connected to their Stripe account.

Expected: 
* [ ] Show an error message "You need to connect to your Stripe account, if you choose Stripe as a payment gateway."
* [ ] While solving this, please also get rid of the empty line between "Yes accept..." and "Stripe accepts Visa..."

Screenshot from 2020-10-07 13-07-58

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes #5262 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/components/events/view/publish-bar.ts  10
- app/mixins/event-wizard.js  29
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6318712&bid=20540279)

File: app/components/events/view/publish-bar.ts [hunk 1]
```diff
@@ -58,6 +63,12 @@ export default class EventsViewPublishBar extends Component<EventsViewPublishBar
             id: 'event_tickets'
           });
         return;
+      } else if (event.canPayByStripe && !(event.stripeAuthorization.stripeAuthCode || event.stripeAuthorization.stripePublishableKey)) {
+        this.notify.error(this.l10n.t('You need to connect to your Stripe account, if you choose Stripe as a payment gateway.'),
+          {
+            id: 'event_stripe'
+          });
+        return;
```
Comment (iamareebjamal) [inline app/components/events/view/publish-bar.ts:71]: Duplicate logic. Move in the event model

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/components/events/view/publish-bar.ts  8
- app/mixins/event-wizard.js  29
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6318712&bid=20540279)

File: app/models/event.js [hunk 2]
```diff
@@ -181,6 +181,10 @@ export default class Event extends ModelBase.extend(CustomPrimaryKeyMixin, {
 
   sessionsByState: computed('sessions', function() {
     return groupBy(this.sessions.toArray(), 'data.state');
+  }),
+
+  stripeConnectCompulsion: computed('canPayByStripe', 'stripeAuthorization.stripeAuthCode', 'stripeAuthorization.stripePublishableKey', function() {
```
Comment (iamareebjamal) [inline app/models/event.js:186]:
```suggestion
  isStripeConnectionValid: computed('canPayByStripe', 'stripeAuthorization.stripeAuthCode', 'stripeAuthorization.stripePublishableKey', function() {
```

Improve variable naming

File: app/models/event.js [hunk 3]
```diff
@@ -181,6 +181,10 @@ export default class Event extends ModelBase.extend(CustomPrimaryKeyMixin, {
 
   sessionsByState: computed('sessions', function() {
     return groupBy(this.sessions.toArray(), 'data.state');
+  }),
+
+  isStripeConnectionValid: computed('canPayByStripe', 'stripeAuthorization.stripeAuthCode', 'stripeAuthorization.stripePublishableKey', function() {
```
Comment (iamareebjamal) [inline app/models/event.js:186]: The name is isStripeConnectionValid and returns true when the connection is invalid :astonished:

File: app/mixins/event-wizard.js [hunk 4]
```diff
@@ -217,14 +217,22 @@ export default Mixin.create(MutableArray, CustomFormMixin, {
     saveDraft() {
       this.onValid(() => {
         preSaveActions.call(this);
-        this.set('data.event.state', 'draft');
-        this.sendAction('save');
+        if (this.data.event.isStripeConnectionValid) {
+          this.notify.error('You need to connect to your Stripe account, if you choose Stripe as a payment gateway.');
```
Comment (iamareebjamal) [inline app/mixins/event-wizard.js:221]: If the connection is valid, show an error?

**Review by iamareebjamal — APPROVED** [review]

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/components/events/view/publish-bar.ts  8
- app/mixins/event-wizard.js  29
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6318712&bid=20540279)

**Pull request merged**